plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/bedrock-runtime.ts

```diff
@@ -579,4 +590,145 @@ export class BedrockRuntimeServiceExtension implements ServiceExtension {
       }
     }
   }
+
+  private async responseHookInvokeModelWithResponseStream(
+    response: NormalizedResponse,
+    span: Span,
+    tracer: Tracer,
+    config: AwsSdkInstrumentationConfig
+  ): Promise<any> {
+    const stream = response.data?.body;
+    const modelId = response.request.commandInput?.modelId;
+    if (!stream) return;
+
+    const wrappedStream =
+      BedrockRuntimeServiceExtension.instrumentAsyncIterable(
+        stream,
+        async (chunk: { chunk?: { bytes?: Uint8Array } }) => {
+          const parsedChunk = BedrockRuntimeServiceExtension.parseChunk(
+            chunk?.chunk?.bytes
+          );
+
+          if (!parsedChunk) return;
+
+          if (modelId.includes('amazon.titan')) {
+            BedrockRuntimeServiceExtension.recordTitanAttributes(
+              parsedChunk,
+              span
+            );
+          } else if (modelId.includes('anthropic.claude')) {
+            BedrockRuntimeServiceExtension.recordClaudeAttributes(
+              parsedChunk,
+              span
+            );
+          } else if (modelId.includes('amazon.nova')) {
+            BedrockRuntimeServiceExtension.recordNovaAttributes(
+              parsedChunk,
+              span
+            );
+          }
+        }
+      );
+    // Replace the original response body with our instrumented stream.
+    // - Defers span.end() until the entire stream is consumed
+    // This ensures downstream consumers still receive the full stream correctly,
+    // while OpenTelemetry can record span attributes from streamed data.
+    response.data.body = (async function* () {
+      try {
+        for await (const item of wrappedStream) {
```

Comment:
IIUC, the result here is that *two* async iterators are created. Could the chunk handling above be moved into this `for await ...` and then not bother having a `wrappedStream` at all? I haven't tried this.

Reply:
Done